src/tapis-hooks/files/useMkdir.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect } from 'react';
+import { useMutation } from 'react-query';
+import { Files } from '@tapis/tapis-typescript';
+import { mkdir } from 'tapis-api/files';
+import { useTapisConfig } from 'tapis-hooks';
+import QueryKeys from './queryKeys';
+
+const useMkdir = (systemId: string, path: string) => {
+  const { basePath, accessToken } = useTapisConfig();
+  const jwt = accessToken?.access_token || '';
+
+  // The useMutation react-query hook is used to call operations that make server-side changes
+  // (Other hooks would be used for data retrieval)
+  //
+  // In this case, mkdir helper is called to perform the operation
+  const { mutate, isLoading, isError, isSuccess, data, error, reset } =
+    useMutation<Files.FileStringResponse, Error, Files.MkdirOperationRequest>(
+      [QueryKeys.mkdir, systemId, path, basePath, jwt],
+      (request: Files.MkdirOperationRequest) => mkdir(request, basePath, jwt)
```

Comment:
onSuccess should be included, that forces a refetch on the systemId and parent dir. This will allow for an automatic refresh of the file listing.